File: src/main/java/com/nilssonmandola/versioningpoc/java/resource/wallet/internal/dto/bitcoin/BitcoinV3DTO.java
package com.nilssonmandola.versioningpoc.java.resource.wallet.internal.dto.bitcoin;

import com.nilssonmandola.versioningpoc.java.resource.wallet.internal.dto.amount.AmountDTO;
import com.nilssonmandola.versioningpoc.java.resource.wallet.internal.dto.bitcoin.kotlin.BitcoinDTO;

public class BitcoinV3DTO implements BitcoinDTO {

    public final String ownerName;
    public final String walletName;
    public final String number;
    public final AmountDTO balance;

    public BitcoinV3DTO(String ownerName, String walletName, String number, AmountDTO balance) {
        this.ownerName = ownerName;
        this.walletName = walletName;
        this.number = number;
        this.balance = balance;
    }
}

File: src/main/java/com/nilssonmandola/versioningpoc/java/resource/wallet/internal/dto/bitcoin/kotlin/BitcoinV3DTO.kt
package com.nilssonmandola.versioningpoc.java.resource.wallet.internal.dto.bitcoin.kotlin

import com.nilssonmandola.versioningpoc.java.resource.wallet.internal.dto.amount.AmountDTO

class BitcoinV3DTO(val ownerName: String,
                   val walletName: String,
                   val number: String,
                   val balance: AmountDTO) : BitcoinDTO